cirq/google/engine/engine_job.py

```diff
@@ -317,6 +325,33 @@ def results(self) -> List[study.Result]:
                     'invalid result proto version: {}'.format(result_type))
         return self._results
 
+    def calibration_results(self):
+        """Returns the results of a run_calibration() call.
+
+        This function will fail if any other type of results were returned
+        by the Engine.
+        """
+        import cirq.google.engine.engine as engine_base
+        if not self._calibration_results:
+            result = self._wait_for_result()
+            result_type = result.type_url[len(engine_base.TYPE_PREFIX):]
+            if result_type != 'cirq.google.api.v2.FocusedCalibrationResult':
+                raise ValueError(
+                    'Did not find calibration results, instead found: {}'.
+                    format(result_type))
+            parsed_val = v2.calibration_pb2.FocusedCalibrationResult.FromString(
+                result.value)
+            cal_results = []
+            for layer in parsed_val.results:
+                metrics = calibration.Calibration(layer.metrics)
+                cal_results.append(
+                    CalibrationResult(
+                        layer.code, layer.error_message, layer.token,
+                        datetime.datetime.fromtimestamp(layer.valid_until_ms /
+                                                        1000), metrics))
```

Comment:
Could you preserve missing values as None, since some parts of the message are optional now?

As alternative I'm fine with keeping them not optional and document which values are used as markers for missing values, I'm not sure what's the convention within Cirq for this.

Reply:
I am not sure if this is possible in proto3.  There is no way to tell if it is unset or the default, see here:
https://docs.google.com/document/d/1oavZD9XB_147ti93MCBoR5HrFKoBh1xkZcTxjInYf0M/edit#heading=h.kozewqqcqhuz
or here for more details:
https://developers.google.com/protocol-buffers/docs/proto3#default

Maybe I should change it back from Optional in the container class?

Reply:
Yes, that seems reasonable. Please document those special values though.

I wonder what's the usual Python convention for this.

Reply:
Changed back to Optionals and included a test to make sure the default values if unset were set to None.